tests/e2e/lib/capture_tcp.py

```
@@ -0,0 +1,53 @@
+            if not chunk:
+                return
+            with self.server.lock:
+                with self.server.output_path.open("ab") as handle:
+                    handle.write(chunk)
+
+
+class CaptureServer(socketserver.ThreadingTCPServer):
+    allow_reuse_address = True
+    daemon_threads = True
+
+    def __init__(self, server_address: tuple[str, int], handler_cls: type[CaptureHandler], output_path: Path):
```

Ruff (0.15.10)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)